typings/embedded_io/audioin.d.ts

```diff
@@ -0,0 +1,51 @@
+/*
+* Copyright (c) 2025 Satoshi Tanaka
+*
+*   This file is part of the Moddable SDK Tools.
+*
+*   The Moddable SDK Tools is free software: you can redistribute it and/or modify
+*   it under the terms of the GNU General Public License as published by
+*   the Free Software Foundation, either version 3 of the License, or
+*   (at your option) any later version.
+*
+*   The Moddable SDK Tools is distributed in the hope that it will be useful,
+*   but WITHOUT ANY WARRANTY; without even the implied warranty of
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*   GNU General Public License for more details.
+*
+*   You should have received a copy of the GNU General Public License
+*   along with the Moddable SDK Tools.  If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+declare module "embedded:io/audio/in" {
+  import { Buffer } from "embedded:io/_common";
+
+  type AudioType = "LPCM";
+
+  interface AudioInOptions {
+    bitsPerSample?: 8 | 16;
+    channels?: 1 | 2;
+    sampleRate?: number;
+    audioType?: AudioType;
+    format?: "buffer";
+  }
+
+  class AudioIn {
+    constructor(options: AudioInOptions & {
+      onReadable?: (this: AudioIn, bytes: number) => void;
+    });
+    read(byteLength?: number): ArrayBuffer
+    read(buffer: Buffer): void;
```

Comment:
Return type is `number` when passing a `Buffer` - see the [`read` method](https://ecmatc53.github.io/spec/web/spec.html#-9-io-class-pattern-read-method) of the IO Class Pattern.